lib/honeybadger/client.ex

```diff
@@ -24,23 +33,25 @@ defmodule Honeybadger.Client do
       metrics: metric
     })
 
-    http_mod.post(
-      client.origin <> @metrics_endpoint, body, client.headers
-    )
+    post(client, @metrics_endpoint, body, http_mod)
   end
 
   def send_notice(%__MODULE__{} = client, metric, http_mod \\ HTTPoison) do
     body = JSON.encode!(metric)
 
-    http_mod.post(
-      client.origin <> @notices_endpoint, body, client.headers
-    )
+    post(client, @notices_endpoint, body, http_mod)
   end
 
-  defp headers(api_key) do
-    [{"Accept", "application/json"},
-     {"Content-Type", "application/json"},
-     {"X-API-Key", api_key}]
+  defp post(client, url, body, http_mod) do
+    proxy = [proxy: client.proxy, proxy_auth: client.proxy_auth]
+
+    case client.proxy do
+      nil -> http_mod.post(client.origin <> url, body, client.headers)
+      _ -> http_mod.post(client.origin <> url, body, client.headers, proxy)
+    end
   end
 
+  defp headers(api_key) do
+    @headers ++ [{"X-API-Key", api_key}]
```

Comment:
Reversing the order of this will make it `O(1)` instead of `O(n)`. In this case `n=2`, but `@headers` may grow.

`[{"X-API-Key", api_key}] ++ @headers`